bsdlib/doc/pdn_extension.rst

```diff
@@ -0,0 +1,174 @@
+It then configures the CID, which creates a PDN.
+This PDN is used to handle data that is sent over the connection.
+
+The example code shows how to create a PDN socket with the protocol family NRF_NPROTO_PDN and the proprietary address family NRF_AF_LTE.
+When the ``nrf_socket`` function returns, a CID has been allocated.
+Next, the created CID must be configured to use the application's APN (using ``nrf_connect``).
+If this operation is successful, the library attempts to activate the associated PDN handle.
+If this activation succeeds, ``nrf_connect`` returns with success.
+
+The file descriptor that is returned can only be used to operate on the PDN socket itself.
+For example, you can use it to set and retrieve socket options, to connect, or to close the PDN socket.
+To use the configured PDN, you must use the associated APN name to identify which PDN to use to transfer data (IP/RAW/DNS data); see `Binding a data socket to an APN`_.
+
+Closing a PDN socket
+********************
+
+.. code-block:: c
+
+   nrf_close(pdn_fd);
+
+The close function for PDN sockets de-registers the associated CID/PDN by deactivating the PDN and removes the configuration of the CID.
+
+Any socket that has been bound to the APN/PDN must be closed before closing the PDN socket.
+Otherwise, it will not have any transport and will start reporting errors.
+
+Binding a data socket to an APN
+*******************************
+
+.. code-block:: c
+
+   tcp_fd = nrf_socket(NRF_AF_INET, NRF_SOCK_STREAM, NRF_IPPROTO_TCP);
+   nrf_setsockopt(tcp_fd, NRF_SOL_SOCKET, NRF_SO_BINDTODEVICE, "custom.apn", strlen("custom.apn"));
+   nrf_bind(...);
+   # Data will now flow over "custom.apn" instead of Default context.
+   nrf_connect(...);
+   nrf_send(tcp_fd, "data", 4, 0);
+
+The example code shows how to create an IPv4 TCP stream socket and bind it to an APN using the NRF_SO_BINDTODEVICE socket option.
+The APN name must be set up and activated prior to binding it; see `Creating a PDN socket`_.
+
+.. note::
+   If the PDN that the socket is being bound to has been deactivated by the network, setting the NRF_SO_BINDTODEVICE option returns an error that the APN parameter is not valid.
+   To fix this error, you must create another PDN socket that is associated with the APN, and bind the data socket to it.
+
+Configuring the type of transport
+*********************************
+
+.. code-block:: c
+
+   nrf_setsockopt(pdn_fd, NRF_SOL_PDN, NRF_SO_PDN_AF, [NRF_AF_INET, NRF_AF_INET6],
+                  sizeof(nrf_sa_family_t) * 2);
+
+You can configure a PDN socket to use a specific internet family type of transport, for example, IPv4 only, IPv6 only, or dual stack.
+To do so, set the socket option NRF_SO_PDN_AF to request a specific type of connection.
+You must do this after creating the socket and before activating it with ``nrf_connect``.
+
+To retrieve a list of what was activated by the network, call ``getsockopt`` again after activation, using the same socket option.
```

Comment:
'what' sounds cryptic here, does it refer to the activated sockets? not sure
```suggestion
To retrieve a list of sockets that have been activated by the network, call ``getsockopt`` again after activation, using the same socket option.
```

Reply:
You are correct @ru-fu, it refers to the list of address families for the given PDN socket.
That is, a "list" (pointer plus length) of `nrf_socket_family_t` structs.

Reply:
Updated to "To retrieve a list of what address families were activated by the network..."